server.js

```diff
@@ -4,20 +4,104 @@ const cors = require("cors");
 const app = express();
 
 app.use(cors());
+app.use(express.json());
 
 const welcomeMessage = {
   id: 0,
   from: "Bart",
   text: "Welcome to CYF chat system!",
 };
 
-//This array is our "data store".
-//We will start with one message in the array.
-//Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+// This array is our "data store".
+// We will start with one message in the array.
+// Note: messages will be lost when Glitch restarts our server.
+const messages = [
+  welcomeMessage,
+  {
+    id: 1,
+    from: "BAki",
+    text: "CAn you see me?",
+  },
+];
 
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+app.get("/messages", (request, response) => {
+  response.send({ messages });
+});
+
+// Send a message
+app.post("/messages", (request, response) => {
+  let newMessage = request.body;
+  if (
+    newMessage.id === "" ||
```

Comment:
Are you trying to validate the `id, from and text` ? if they don't exist it will be a bad request right?

your logic check only if the parameter is an empty string, but what if the caller gives an undefined value?